rapids-4-spark-tools/README.md

```diff
@@ -1,78 +1,197 @@
 # Spark Qualification and Profiling tools
 
-The qualification tool is used to rank a set of applications to determine if the RAPIDS Accelerator for Apache Spark is a good fit for those applications.
+The qualification tool is used to rank a set of applications to determine if the RAPIDS Accelerator for Apache Spark 
+is a good fit for those applications. Works with CPU generated event logs.
 
 The profiling tool generates information which can be used for ranking applications debugging and profiling.
-Information such as Spark version, executor information, Properties and so on.  Works with both cpu and gpu generated event logs.
+Information such as Spark version, executor information, properties and so on.  
+Works with both CPU and GPU generated event logs.
 
 (The code is based on Apache Spark 3.1.1 source code, and tested using Spark 3.0.x and 3.1.1 event logs)
 
 ## Prerequisites
-1. Require Spark 3.1.1 or newer installed
+- Spark 3.1.1 or newer installed
+- Java 8 or above
+- Complete Spark event log(s) from Spark 3.0 or above version.
 
-## How to compile
-1. `mvn clean package`
+Optional:
+- maven installed 
+  (only if you want to compile the jar yourself)
+- A running Spark cluster 
+  (only if you choose to run the job on a Spark cluster)
+- hadoop-aws-<version>.jar and aws-java-sdk-<version>.jar 
+  (only if any input event log is from S3)
+  
+## Download the jar or compile it
+You do not need to compile the jar yourself because you can download it from maven repository directly.
+
+Here are 2 options:
+1. Download the jar file from maven repository
+
+2. Compile the jar from github repo
+```bash
+git clone https://github.com/NVIDIA/spark-rapids.git
+cd spark-rapids
+mvn -pl .,rapids-4-spark-tools clean verify -DskipTests
+```
+The jar is generated in below directory :
+
+`./rapids-4-spark-tools/target/rapids-4-spark-tools_2.12-<version>.jar`
+
+## How to use this tool
+
+This tool parses the spark event log(s) and creates an output report.
+Acceptable input spark event log paths are local filesystem, hdfs , S3 or mixed.
+It can be file path , directory path or mixed.
+
+Below is an example input:
+```
+/path/to/localfilesystem/eventlog1 hdfs://path/to/directory/ s3a://<BUCKET>/to/directory/
+```
+
+If any input is a S3 file path or directory path, here 2 extra steps to access S3 in Spark:
+1. Download the matched jars based on the Hadoop version:
+   - hadoop-aws-<version>.jar
+   - aws-java-sdk-<version>.jar 
+     
+Take Hadoop 2.7.4 for example, we can download and include below jars in the '--jars' option to spark-shell or spark-submit:
+[hadoop-aws-2.7.4.jar](https://repo.maven.apache.org/maven2/org/apache/hadoop/hadoop-aws/2.7.4/hadoop-aws-2.7.4.jar) and 
+[aws-java-sdk-1.7.4.jar](https://repo.maven.apache.org/maven2/com/amazonaws/aws-java-sdk/1.7.4/aws-java-sdk-1.7.4.jar)
+
+2. In $SPARK_HOME/conf, create `hdfs-site.xml` with below AWS S3 keys inside:
```

Comment:
si there easier way to do this to send to specific job?  Many times modifying /conf isn't an option on shared cluster. I thought there was env variables?

Reply:
I tried env variable and it does not work.
Only this way works.